src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/FilterDelegationBase.cs

```diff
@@ -75,78 +76,97 @@ protected bool IsValidDelegatableFilterPredicateNode(TexlNode dsNode, TexlBindin
             NodeKind kind;
             kind = dsNode.Kind;
 
-            switch (kind)
+            ErrorContainer originalErrorContainer = null;
+            try
             {
-                case NodeKind.BinaryOp:
-                    {
-                        var opNode = dsNode.AsBinaryOp();
-                        var binaryOpNodeValidationStrategy = GetOpDelegationStrategy(opNode.Op, opNode);
-                        Contracts.AssertValue(opNode);
+                // if hints should not be generated, create a temporary error container
+                if (!generateHints)
+                {
+                    originalErrorContainer = binding.ErrorContainer;
+                    binding.ErrorContainer = new ErrorContainer();
```

Comment:
How this would be used? I am little concern about making errorcontainer property mutable.

Reply:
The scenario is that we want to allow delegation validation in the short term that does not result in warnings shown to the end-user, purely for telemetry.  The current validation code directly adds warnings to the ErrorContainer on the binder.  A previous PR (https://github.com/microsoft/Power-Fx/pull/914) changed the logic to limit emitting warnings, but it was a more significant code change that was fragile.